pilot/pkg/xds/ads.go

```diff
@@ -470,10 +466,6 @@ func (s *DiscoveryServer) initConnection(node *core.Node, con *Connection) error
 
 	s.addCon(con.ConID, con)
 
-	if s.InternalGen != nil {
-		s.InternalGen.OnConnect(con)
```

Comment:
I agree with splitting this but where is this stuff being called now?

Reply:
+1 – it's unclear why a lot of the `OnConnect` and `startPush` logic was removed/moved up.

Reply:
As we can see, the `OnConnect` would send a real response to sidecar, which sidecar actually do not care about. It wastes both cpu and bandwidth. In general, for xds all resources should be subscribed before generating/pushing responses for them

Reply:
We could bring back onConnect/onDisconnect callback when necessary, but could not understand by what means they should belong to internal generator.

Reply:
The design is to have debugging client (istioctl) connect to internal gen and get info about connections, disconnects, etc. The implementation is not fully complete/tested, but I don't think we should just fully remove it

Reply:
For these, We have /debug restful api exposed now

Reply:
I don't think it is good to send useless info to normal proxies. It is garbage for them.

Reply:
It doesn't send it to the normal proxies. We only send the configs the request

Reply:
Right, then i am curious what's the use case here

>The design is to have debugging client (istioctl) connect to internal gen

This pr does not change this.

Reply:
Yes it does, since we don't trigger on connect anymore which notifies the debugging client when a new proxy connects

Reply:
I see @howardjohn's point. I also think internalgen is little confusing refactored a little bit here https://github.com/istio/istio/pull/29705. PTAL. I Am OK to revert renaming if internalgen is more appropriate for this.